server/src/main/java/org/elasticsearch/search/aggregations/AggregationErrors.java

```diff
@@ -21,4 +27,89 @@ private AggregationErrors() {}
     public static IllegalArgumentException invalidPathElement(String element, String path) {
         return new IllegalArgumentException("Invalid path element [" + element + "] in path [" + path + "]");
     }
+
+    /**
+     * This error indicates that an aggregation is being used on a value type that it is not capable of processing, such as taking
+     * the sum of a keyword. It is a 400 class error and should not be retried.
+     *
+     * This is a legacy version of this error; in general, we should aim to use the
+     * {@link org.elasticsearch.search.aggregations.support.ValuesSourceType} version below
+     *
+     * @param valuesSource The values source we resolved from the query
+     * @param name The name of the aggregation
+     * @return an appropriate exception type
+     */
+    public static IllegalArgumentException unsupportedValuesSourceType(ValuesSource valuesSource, String name) {
+        return new IllegalArgumentException(
```

Comment:
It might be better to use something that extends from `ElasticsearchException` and specifically overrides the `status()` method, but that's not what we've done historically so I just followed the pattern we've used before.